### PR TITLE
darwinbuild-legacy: convert to github portgroup

### PR DESCRIPTION
**Pull request by kencu**
Darwin8 and Darwin9 only

###### Tested on
10.4 PPC and 10.5 PPC

### CODE REVIEW

File: devel/darwinbuild-legacy/Portfile [hunk 1]
```diff
@@ -1,53 +1,51 @@
 PortSystem      1.0
+PortGroup       github 1.0
 
 name            darwinbuild-legacy
 version         0.8.0
-revision	655
```
Comment (mojca) [inline devel/darwinbuild-legacy/Portfile]: You are effectively lowering the version (except that `github.setup` probably overrides the version field). Either version or revision should be higher that it was.

File: devel/darwinbuild-legacy/Portfile [hunk 2]
```diff
@@ -1,53 +1,51 @@
...
 categories      devel
 platforms       darwin
+license         BSD
 maintainers     wms
 description     Darwinbuild is a set of tools that facilitate building the sources \
-		released by Apple.
+                released by Apple.
```
Comment (mojca) [inline devel/darwinbuild-legacy/Portfile]: Whitespace changes should usually go to a separate commit from content changes.

File: devel/darwinbuild-legacy/Portfile [hunk 3]
```diff
@@ -1,53 +1,51 @@
...
-			    10 and later, install the darwinbuild port." 
+                10 and later, install the darwinbuild port."
     }
 }
 
+
 platform darwin 9 {
-	worksrcdir	Darwin9
-	svn.url         ${master_sites}/releases/${worksrcdir}
+    github.setup    macosforge darwinbuild 4848b7f5b223e094a26c413b451d27b938b0f2bd
```
Comment (mojca) [inline devel/darwinbuild-legacy/Portfile]: Doesn't this override the version? You should probably set the version after `github.setup`.

**Comment by kencu** [comment]
Hi, Mojca -- I think this port originally used the revision field to represent the svn revision for the old svn fetch, as that seemed to correlate with the markers I could see in the various commits. It's not really a MacPorts revision. 

The actual code of the listed version (0.8.0) hasn't changed for a number years - and it probably never will change, I would think, for Darwin8 or Darwin9. I picked the last commit from each of the Darwin8 and Darwin9 branches of darwinbuild for the github.setup.

I guess I could set the revision at 655, as it was before. It wouldn't really matter much. If anyone actually has this port installed, I don't want to force a rebuild. And for any new installs (which is what I was aiming for) it wouldn't really matter what the revision was.

The reason for updating this port was to install darwinbuild, which Jeremy had suggested be working for a new libgcc_s.10.5.dylib port for 10.4 and 10.5 I'm working on with him.

**Comment by kencu** [comment]
The whitespace - you're right. I just fixed it up all at once, but I'll try to split that up for the future.

I think you're quite right. I'd have to set the version after the github.setup, or the version will get overridden by that. Goo point.

**Comment by mojca** [comment]
Yes, I can imagine that the revision was being used for the actual SVN revision, but it was still used for the port versioning and I still don't think that we should be ever decreasing it unless you also change the epoch (and then remove the revision).

Upgrading the port doesn't sound so horribly bad either even if the contents didn't change, but if you are 100% sure that nothing changed, just leave the revision at current value (no need to increase it).

**Comment by mojca** [comment]
* The commit message should reference the Trac tickets.
* We need to file a port abandoned ticket. In case the maintainer doesn't respond (I have never heard of him, don't find any emails in my personal archive, he hasn't made a link to github), are you potentially willing to take it over?

**Comment by kencu** [comment]
Yep, can do.

BTW, I also did fix the darwinbuild port, see 52355. 

There was a 64->32 bit integer error building for 64bit that nobody else seemed  to run into, so I had to patch the source to fix up the int size in a few places. I'd like another pair of eyes to look that over.

**Comment by kencu** [comment]
I thought I might as well post up the build and test use logs for Tiger and Leopard PPC for anyone who wants to see that they installed worked, as not too many of us have these machines to actually verify that these ports install as I say they do.

https://pastebin.com/SMyJ1Std
https://pastebin.com/axyAFq6P

**Comment by mojca** [comment]
Why did you close the PR?

File: devel/darwinbuild-legacy/Portfile [hunk 4]
```diff
@@ -1,53 +1,55 @@
...
 
 long_description \
-		The Darwin Build Scripts are a collection of tools that assist compilation of \
-		the many projects contained in Darwin, the open source base of Apple's \
-		Mac OS X operating system.  Apple publishes the sources of these projects \
-		in an archive format (.tar.gz).  An archive is published for each project \
-		version on Apple's site. These tools will provide the proper build environment as well as help to \
-		resolve any necessary dependencies prior to building.
-
-homepage        http://darwinbuild.macosforge.org/
```
Comment (mojca) [inline devel/darwinbuild-legacy/Portfile]: Homepage is still needed. It seems to be at https://macosforge.github.io/darwinbuild/.
(But don't worry adding it now, I'm in the middle of committing.)

**Comment by mojca** [comment]
I'm sorry, I screwed up the merge process.
c6d1c050e16e12ac518fd3c66b423e84069d1be1 and fc2db92a65e9968a83b9c7532c9372ad3a75fcaa

Anyway, please try to split the commit into multiple parts yourself next time when you submit a PR. If someone else needs to do it it's about quadruple work.

**Comment by kencu** [comment]
Will do - thank you! - I certainly don't want to increase anyone's work. Indeed I thought the github portgroup would add in the homepage as it usually does. Now I know I can't count on that.

I was quite impressed with Sterling Smith's note <https://lists.macports.org/pipermail/macports-dev/2017-May/035830.html> that adding ?w=1 onto the web address when viewing the commit makes all the whitespace changes magically disappear and you see pretty much only the content change. Very cool. But no need to make people do this all the time, for sure.
